scipy/ndimage/tests/test_filters.py

```diff
@@ -1079,8 +1140,11 @@ def test_filter_tuple_length_mismatch(self, n_mismatch, filter_func,
         # Test for the intended RuntimeError when a kwargs has an invalid size
         array = xp.arange(6 * 8 * 12, dtype=xp.float64)
         array = xp.reshape(array, (6, 8, 12))
-        kwargs = dict(**kwargs, axes=(0, 1))
+        axes = (0, 1)
+        kwargs = dict(**kwargs, axes=axes)
         kwargs[key] = (val,) * n_mismatch
+        if filter_func in [ndimage.convolve, ndimage.correlate]:
+            kwargs["weights"] = np.ones((5,) * len(axes))
```

Comment:
`weights` is an array, so we want `xp.ones` here rather than `np.ones` - that's causing the CI failures

Reply:
thanks, updated